mobile/lib/pages/editing/edit.page.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:immich_mobile/pages/editing/crop.page.dart';
+import 'package:immich_mobile/entities/asset.entity.dart';
+import 'package:immich_mobile/widgets/common/immich_image.dart';
+import 'package:immich_mobile/widgets/common/immich_toast.dart';
+import 'package:immich_mobile/providers/api.provider.dart';
+import 'package:immich_mobile/services/api.service.dart';
+
+@immutable
+class EditImagePage extends StatelessWidget {
+  final dynamic imageSource; // Can be either Asset or Image
```

Comment:
I'd prefer using a nullable:

```dart
final Asset? asset;
final Image? image;
```

And then in the constructor, you can add an `assert`

```dart
const EditImagePage({
  super.key,
  this.image,
  this.asset,
}) : assert((this.image != null && this.asset == null) || (this.image == null && this.asset != null)), 'Must supply one of asset or image');
```

It's a little hacky, but at least it lets us avoid using typeless `dynamic` variables.

Reply:
It was lot of hacky lol but got it changed in the code!